_ingest-pipelines/processors/script.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: default
+title: script
+parent: Ingest processors
+nav_order: 230
+---
+
+# `script` processor
+
+The `script` processor executes inline and stored scripts that can modify or transform data in an OpenSearch document during the ingestion process. The processor uses script caching for improved performance, as scripts may be recompiled per document. Refer to [Script APIs](https://opensearch.org/docs/latest/api-reference/script-apis/index/) for information about working with scripts in OpenSearch. 
+
+The following is the syntax for the `script` processor:
+
+```json
+{
+  "processor": {
+    "script": {
+      "tag": "<processor_tag>",
+      "description": "<processor_description>",
+      "source": "<script_source>",
+      "lang": "<script_language>",
+      "params": {
+        "<param_name>": "<param_value>"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `script` processor.
+
+| Parameter  | Required/Optional  | Description  |
+|---|---|---|
+`source`  | Required  | The Painless script to be executed.
```

Comment:
`source`, `lang` and `params` are optional, and another optional parameter `id` is missing, `id` can be used to reference a stored script which was created by this API:https://opensearch.org/docs/latest/api-reference/script-apis/create-stored-script/, `source` and `id` are mutually exclusive, either of them can be set.